### DISCUSSION TITLE
Add built-in server monitoring

### DISCUSSION THREAD

**Discussion by jaydrogers** [serversideup/spin#58] [Ideas]
# 👉 Describe the problem
[comment]: <> (What are you trying to solve?)
- There isn't any monitoring built in with Spin

# 👥 Problem evidence & reach
[comment]: <> (How many people have this problem?)
- Discussion in Discord: https://discord.com/channels/910287105714954251/910299290230997003/1195017489273327636

# 🥰 Describe the "impact" on users?
[comment]: <> (How will this make people's lives better once it is solved?)
- Users will be happy they can trust their server is working correctly

# 🏆 How to solve this problem
[comment]: <> (Describe the feature that you are proposing and how it will solve the problem)

### Add a new role to the Ansible Collection
- [ ] Create a monitoring role with Ansible (https://github.com/serversideup/ansible-collection-spin)

### Determine which monitoring to support
> [!IMPORTANT]  
> **🗣️This section is still up for discussion.** Please comment below if you'd like to see something added!

#### Netdata (https://www.netdata.cloud/)
- Netdata is something that can be installed on the machine and easily connected to their cloud
- Downfall is they are an open source, but VC funded company. Not sure how people would feel about this

#### Prometheus/Grafana (https://prometheus.io/) (https://grafana.com/)
- This is a full open source solution, but we would need to fully understand what overhead this would add to the server

# 💯 How do we validate the problem is solved?
[comment]: <> (Explain use cases on how we can measure the success of this implementation)
- [ ] Users are able to provision a server with email or slack alerts 
- [ ] Users should be able to get started without provisioning additional infrastructure for a single server
- [ ] Users should be able to choose if they want monitoring enabled or not

**Comment by NathanLochala** [comment]
@jaydrogers Has there been any progress with this? I had a redis container crash on me due to an AOF corruption file. I didn't know about it apart from noticing all my scheduled jobs failed to run. I would LOVE to see some monitoring built into SPIN.

**Reply by jaydrogers** [reply]
No plans yet. This only has 4 upvotes and we have other feature requests in front of it in regards to other projects.

If time is of the essence, I'd suggest implementing the solutions we have listed in the feature request above.
